RUFAS/routines/field/crop/crop_data_factory.py

```diff
@@ -0,0 +1,203 @@
+from typing import Any, TypedDict
+
+from RUFAS.data_structures.crop_soil_to_feed_storage_connection import (
+    CropCategory,
+    CropType,
+    CROP_CATEGORY_TO_CROP_TYPE_MAPPING,
+    StorageType,
+)
+from RUFAS.data_structures.feed_storage_to_animal_connection import RUFAS_ID
+from RUFAS.input_manager import InputManager
+from RUFAS.output_manager import OutputManager
+
+from .crop_data import CropData, PlantCategory
+
+
+class CropConfiguration(TypedDict):
+    """
+    Data structure used to store crop configuration attributes. Attribute descriptions are omitted because all
+    attributes are documented in both the metadata properties and the CropData class docstring.
+    """
+
+    name: str
+    plant_category: PlantCategory
+    crop_category: CropCategory
+    crop_type: CropType
+    rufas_ids: list[RUFAS_ID]
+    is_nitrogen_fixer: bool
+    minimum_temperature: float
+    optimal_temperature: float
+    potential_heat_units: float
+    max_leaf_area_index: float
+    first_heat_fraction_point: float
+    first_leaf_fraction_point: float
+    second_heat_fraction_point: float
+    second_leaf_fraction_point: float
+    senescent_heat_fraction: float
+    light_use_efficiency: float
+    emergence_nitrogen_fraction: float
+    half_mature_nitrogen_fraction: float
+    mature_nitrogen_fraction: float
+    emergence_phosphorus_fraction: float
+    half_mature_phosphorus_fraction: float
+    mature_phosphorus_fraction: float
+    max_root_depth: float
+    root_distribution_param_da: float
+    root_distribution_param_c: float
+    storage_type: StorageType
+    optimal_harvest_index: float
+    minimum_harvest_index: float
+    dry_matter_percentage: float
+    lignin_dry_matter_percentage: float
+    crude_protein_percent: float
+    non_protein_nitrogen: float
+    starch: float
+    adf: float
+    ndf: float
+    sugar: float
+    ash: float
+    yield_nitrogen_fraction: float
+    yield_phosphorus_fraction: float
+
+
+class CropDataFactory:
+    """
+    Manages and manufactures CropData instances using user-input crop configurations.
+
+    Attributes
+    ----------
+    _crop_configurations : dict[str, dict[str, CropConfiguration]]
```

Comment:
```suggestion
    _crop_configurations : dict[str, CropConfiguration]
```
I believe it's correct in the attribute type hint but regardless they should match.

Reply:
Good catch